components/usage/pkg/apiv1/billing_test.go

```diff
@@ -93,26 +146,52 @@ func TestCreditSummaryForTeams(t *testing.T) {
 					AttributionId: string(teamAttributionID_A),
 					Credits:       (12) * 10,
 					StartTime:     timestamppb.New(time.Now().AddDate(0, 0, -1)),
+					TeamId:        string(teamAttributionID_A),
 				},
 				{
 					// has 1 day of usage, but started three days ago
 					AttributionId: string(teamAttributionID_A),
 					Credits:       (24) * 10,
 					StartTime:     timestamppb.New(time.Now().AddDate(0, 0, -3)),
+					TeamId:        string(teamAttributionID_A),
 				},
 			},
-			Expected: map[string]int64{
-				teamID_A: 120,
+			Expected: map[string]map[string]float64{
+				teamID_A: {
+					"creditsUsed":     120,
+					"spendingLimit":   float64(read.SpendingLimit),
+					"upcomingInvoice": 130,
+				},
 			},
 		},
 	}
 
 	for _, s := range scenarios {
 		t.Run(s.Name, func(t *testing.T) {
-			svc := NewBillingService(&stripe.Client{}, s.BillSessionsAfter, &gorm.DB{})
-			actual, err := svc.creditSummaryForTeams(s.Sessions)
+			svc := NewBillingService(&testStripeClient{}, s.BillSessionsAfter, &gorm.DB{}, usageClient)
+			actual, err := svc.creditSummaryForTeams(ctx, s.Sessions)
 			require.NoError(t, err)
-			require.Equal(t, s.Expected, actual)
+			require.Equal(t, s.Expected["creditsUsed"], actual["creditsUsed"])
 		})
 	}
+	t.Cleanup(func() {
+		dbconn.Model(&db.CostCenter{}).Delete(costCenter, costCenter2)
+	})
+}
+
+type testStripeClient struct{}
```

Comment:
Referred to [here](https://github.com/gitpod-io/gitpod/pull/12382/files#r956462065). Would love to learn how to mock this better.